### PR TITLE
Add-remove-edit functions

### PR DESCRIPTION
**Pull request by Boku52500**
- Created add function to add tasks into list (with enter button)
- Created remove function to remove tasks from the list (with trash button)
- Created edit function to edit tasks within the list by just clicking on it.
- Used local storage to store data.
- Used an empty array to store the tasks.
- Fixed linters.

### CODE REVIEW

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @Boku52500 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights :heavy_check_mark: 
- [ ] Nice UI design :white_check_mark: 
- [ ] Professional readme file :white_check_mark: 
- [ ] Descriptive pull request title and summary :white_check_mark: 
- [ ] Good descriptive commit messages :white_check_mark: 

almost-there

## Required Changes ♻️
- [x] Please consider adding the mode of the application to the `webpack.config.js` file, to get rid of the error in the screenshot below. 

Screenshot from 2022-11-05 01-56-32

You can solve this by adding the `mode` configuration as illustrated below
```JavaScript
  module.exports = {
  mode: 'production',
  entry: './src/index.js',
  plugins: [
    new HtmlWebpackPlugin({
      template: './src/index.html',
    }),
  ],
  output: {
    filename: 'main.js',
    path: path.resolve(__dirname, 'dist'),
  },
  devServer: {
    static: './dist',
  },
  module: {
    rules: [
      {
        test: /\.css$/i,
        use: ['style-loader', 'css-loader'],
      },
      {
        test: /\.(png|svg|jpg|jpeg|gif)$/i,
        type: 'asset/resource',
      },
    ],
  },
};
```

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/actions.js [hunk 1]
```diff
@@ -1,52 +1,78 @@
...
+    } else {
+      tasks = [];
+      const input = document.querySelector('.list-create input').value;
+      const completed = false;
+      const index = 0;
+      const task = new Todo(input, completed, index);
+      tasks.push(task);
+      Actions.sincronizationStorage();
+    }
+  }
```
Comment (MaganaAsiati) [inline modules/actions.js:36]: - You did a great job, your design looks nice but make sure you can be able to add tasks, remove tasks, make them complete, add them to the local storage and clear all, on my side that functionality is not working as expected kindly fix that. :+1:

**Review by MaganaAsiati — CHANGES_REQUESTED** [review]
# STATUS: Required Changes ♻️♻️♻️ 
## Hi @jossi30, I'm @MaganaAsiati, your reviewer for this PR  :wave: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close-gif-4

## Highlights

- Descriptive README file  :heavy_check_mark:👨🏻‍💻
- Great work using the right GitHub Flow.:heavy_check_mark:
- No Linter errors :heavy_check_mark:

## Required Changes ♻️

- [ ] Kindly make sure you implement the functionality I list below as it was required in the project  :+1: 

```
- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for editing task descriptions.
- By default new tasks should have the property `completed` set to `false` and the property `index` set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
- All changes to the To Do List should be saved in local storage.

```

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
- [ ] [OPTIONAL] Please make sure to make your commit messages more descriptive in the future. :heavy_check_mark:👨🏻‍💻
- [ ] kindly  Add a meaningful description to your pull request in the future which gives more information about what you implement in this milestone :heavy_check_mark:👨🏻‍💻


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by Boku52500** [comment]
> # STATUS: Required Changes 
> ## Hi @jossi30, I'm @MaganaAsiati, your reviewer for this PR 
> Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
> 
> so-close-gif-4 [ ](https://user-images.githubusercontent.com/95297251/191952218-b7185270-c02b-4e4d-89af-eae77fcd8bea.gif) so-close-gif-4 [ ](https://user-images.githubusercontent.com/95297251/191952218-b7185270-c02b-4e4d-89af-eae77fcd8bea.gif)
> 
> ## Highlights
> * Descriptive README file  heavy_check_mark👨🏻‍💻
> * Great work using the right GitHub Flow.heavy_check_mark
> * No Linter errors heavy_check_mark
> 
> ## Required Changes 
> * [ ]  Kindly make sure you implement the functionality I list below as it was required in the project  +1
> 
> ```
> - Remove all hardcoded items from the tasks array.
> - Create a new JavaScript file for the new functionality.
> - Implement a function for adding a new task (add a new element to the array).
> - Implement a function for deleting a task (remove an element from the array).
> - Implement a function for editing task descriptions.
> - By default new tasks should have the property `completed` set to `false` and the property `index` set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
> - Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
> - All changes to the To Do List should be saved in local storage.
> ```
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
> 
> * [ ]  [OPTIONAL] Please make sure to make your commit messages more descriptive in the future. heavy_check_mark👨🏻‍💻
> * [ ]  kindly  Add a meaningful description to your pull request in the future which gives more information about what you implement in this milestone heavy_check_mark👨🏻‍💻
> 
> Cheers and Happy coding!clapclapclap
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Hello, sorry I think you reviewed the wrong branch, because I already have everything implemented in given list

File: modules/actions.js (hunk 1 above)
Comment (Boku52500) [reply modules/actions.js:36]: Hello, I am sorry,  I think you are reviewing the wrong branch. I already implemented everything in this branch from the given list.

**Review by mgmediaweb — APPROVED** [review]
Hi @Boku52500 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- Good job following the advice of the previous reviewer ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**